internal/pkgecosystem/pypi.go

```diff
@@ -3,13 +3,27 @@ package pkgecosystem
 import (
 	"encoding/json"
 	"fmt"
+	"io"
 	"net/http"
+	"strings"
 )
 
-type pypiJSON struct {
+// pyPIPackageInfoJSON represents relevant JSON data from the PyPI web API response
+// when package information is requested. The differences in response format between
+// (valid) requests made with a specific package version and with no package version
+// are not significant in our case.
+// (In particular, if the request contains a valid version, Urls contains a single entry
+// holding information for that package version. If the version is unspecified, Urls contains
+// an entry corresponding to each version of the package available on PyPI.)
+// See https://warehouse.pypa.io/api-reference/json.html and https://peps.python.org/pep-0691
+type pyPIPackageInfoJSON struct {
```

Comment:
nit: for consistency with the below struct, just call this "pypiPackageInfoJSON"